bin/install

```diff
@@ -20,7 +20,7 @@ get_maven() {
 	local destdir=$2
 
 	local major=$(echo $version | cut -d '.' -f 1)
-	local base="https://archive.apache.org/dist/maven"
+	local base="${ASDF_MAVEN_MIRROR:=https://archive.apache.org/dist/maven}"
 	local url="$base/maven-$major/$version/binaries/apache-maven-$version-bin.tar.gz"
```

Comment:
Using [`closer.lua`](https://infra.apache.org/release-download-pages.html#closer) would help direct traffic to CDN for releases available there ([example](https://www.apache.org/dyn/closer.lua/maven/maven-3/3.9.9/binaries/apache-maven-3.9.9-bin.tar.gz)).  For older releases, it redirects to archive ([example](https://www.apache.org/dyn/closer.lua/maven/maven-3/3.6.3/binaries/apache-maven-3.6.3-bin.tar.gz)).

```suggestion
  local path="maven-$major/$version/binaries/apache-maven-$version-bin.tar.gz"
  local url="https://www.apache.org/dyn/closer.lua/maven/$path?action=download"
  if [[ -n "${ASDF_MAVEN_MIRROR:-}" ]]; then
    url="${ASDF_MAVEN_MIRROR}/$path"
  fi
```